jms/src/main/scala/akka/stream/alpakka/jms/JmsConnector.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.jms
+
+import javax.jms
+import javax.jms.{ ExceptionListener, JMSException }
+
+import akka.stream.ActorMaterializer
+import akka.stream.stage.GraphStageLogic
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+/**
+ * Internal API
+ */
+private[jms] trait JmsConnector { this: GraphStageLogic =>
+
+  implicit private[jms] var ec: ExecutionContext = _
+
+  private[jms] var jmsSession: JmsSession = _
+
+  private[jms] def jmsSettings: JmsSettings
+
+  private[jms] def onSessionOpened(): Unit = {}
+
+  private[jms] def fail = getAsyncCallback[Throwable](e => failStage(e))
+
+  private def onSession =
+    getAsyncCallback[JmsSession](session => {
+      jmsSession = session
+      onSessionOpened()
+    })
+
+  private[jms] def initSessionAsync() = {
+    ec = materializer match {
+      case m: ActorMaterializer => m.system.dispatchers.lookup("akka.stream.default-blocking-io-dispatcher")
```

Comment:
One important nitpick; this should be taken from the Attributes so that it can be overridden.

In `createLogic` you can grab the dispatcher id with something like `inheritedAttributes.get[ActorAttributes.Dispatcher](Dispatcher("akka.stream.default-blocking-io-dispatcher"))`